### PR TITLE
Fix mobile search issues

### PR DESCRIPTION
**Pull request by buberdds**
https://app.clickup.com/t/8694h28bx

current master:

Screenshot from 2024-05-14 09-13-15
Screenshot from 2024-05-14 09-12-45

### CODE REVIEW

**Comment by lubej** [comment]
Should we just remove the search in footer on HomePage for mobile? I don't see the point, because you have the search on the top of the page.

**Comment by buberdds** [comment]
I have asked the same question to Don, but an answer was to fix z-index. I will validate this one more time with him.

**Comment by buberdds** [comment]
cc @donouwens

**Comment by lubej** [comment]
> The reason we have got the search in the footer is that inner pages, like block detail or account pages, often get very long. This way the user has an option to access search from the bottom of the page as well. A sticky navigation was considered but deemed to take up too much screen estate.

Yes, makes sense on other pages. But not on home page, where the page's height isn't large, and you can almost seen both search bars at the same time.

File: src/app/components/Search/index.tsx [hunk 1]
```diff
@@ -52,6 +52,9 @@ const SearchForm = styled('form', {
               display: 'none',
             },
             [`.${inputAdornmentClasses.positionEnd}`]: {
+              'button[type="button"]': {
+                display: 'none',
+              },
```
Comment (lubej) [inline src/app/components/Search/index.tsx:57]: Why is this change necessary?

**Review by lubej — APPROVED** [review]

File: src/app/components/Search/index.tsx (hunk 1 above)
Comment (buberdds) [reply src/app/components/Search/index.tsx:57]: (the first screenshot in the description) 
When mobile footer search input is collapsed, but has a value, a part of it is not fully hidden. Search icon positioning is affected too and only half of icon is visible.
